diff -up pciutils-3.0.0/lib/configure.multilib pciutils-3.0.0/lib/configure
--- pciutils-3.0.0/lib/configure.multilib	2008-04-10 21:15:47.000000000 +0200
+++ pciutils-3.0.0/lib/configure	2008-06-02 06:46:35.000000000 +0200
@@ -41,28 +41,61 @@ sys=`echo $host | sed 's/^\([^-]*\)-\([^
 echo " $host $rel"
 
 c=config.h
+cm=config.h.mk
 m=config.mk
-echo >$c "#define PCI_ARCH_`echo $cpu | tr '[a-z]' '[A-Z]'`"
-echo >>$c "#define PCI_OS_`echo $sys | tr '[a-z]' '[A-Z]'`"
 echo >$m 'WITH_LIBS='
+cat >$c <<EOF
+#if defined(__x86_64__)
+#define PCI_ARCH_X86_64
+#elif defined(__ia64__)
+#define PCI_ARCH_IA64
+#elif defined(__i386__)
+#define PCI_ARCH_I386
+#define PCI_HAVE_PM_INTEL_CONF
+#elif defined(__ppc64__) || defined(__powerpc64__)
+#define PCI_ARCH_PPC64
+#elif defined(__ppc__)  || defined(__powerpc__)
+#define PCI_ARCH_PPC
+#elif defined(__s390x__)
+define PCI_ARCH_S390X
+#elif defined(__s390__)
+#define PCI_ARCH_S390
+#elif defined(__alpha__)
+#define PCI_ARCH_ALPHA
+#else
+#error Unknown Arch
+#endif
+#define PCI_OS_LINUX
+#define PCI_HAVE_PM_LINUX_SYSFS
+#define PCI_HAVE_PM_LINUX_PROC
+#define PCI_HAVE_LINUX_BYTEORDER_H
+#define PCI_PATH_PROC_BUS_PCI "/proc/bus/pci"
+#define PCI_PATH_SYS_BUS_PCI "/sys/bus/pci"
+#define PCI_HAVE_64BIT_ADDRESS
+EOF
+
+rm -f $cm
+echo >$cm "#define PCI_ARCH_`echo $cpu | tr 'a-z' 'A-Z'`"
+echo >>$cm "#define PCI_OS_`echo $sys | tr 'a-z' 'A-Z'`"
 
 echo_n "Looking for access methods..."
+echo_n " sysfs proc"
 
 case $sys in
 	linux*)
 		echo_n " sysfs proc"
-		echo >>$c '#define PCI_HAVE_PM_LINUX_SYSFS'
-		echo >>$c '#define PCI_HAVE_PM_LINUX_PROC'
-		echo >>$c '#define PCI_HAVE_LINUX_BYTEORDER_H'
-		echo >>$c '#define PCI_PATH_PROC_BUS_PCI "/proc/bus/pci"'
-		echo >>$c '#define PCI_PATH_SYS_BUS_PCI "/sys/bus/pci"'
+		echo >>$cm '#define PCI_HAVE_PM_LINUX_SYSFS'
+		echo >>$cm '#define PCI_HAVE_PM_LINUX_PROC'
+		echo >>$cm '#define PCI_HAVE_LINUX_BYTEORDER_H'
+		echo >>$cm '#define PCI_PATH_PROC_BUS_PCI "/proc/bus/pci"'
+		echo >>$cm '#define PCI_PATH_SYS_BUS_PCI "/sys/bus/pci"'
 		case $cpu in
 				i386|x86_64)	echo_n " i386-ports"
-						echo >>$c '#define PCI_HAVE_PM_INTEL_CONF'
+						echo >>$cm '#define PCI_HAVE_PM_INTEL_CONF'
 						;;
 		esac
-		echo >>$c '#define PCI_HAVE_64BIT_ADDRESS'
-		echo >>$c '#define PCI_USE_DNS'
+		echo >>$cm '#define PCI_HAVE_64BIT_ADDRESS'
+		echo >>$cm '#define PCI_USE_DNS'
 		;;
 	sunos)
 		case $cpu in
@@ -110,7 +143,7 @@ case $sys in
 		;;
 esac
 
-echo >>$c '#define PCI_HAVE_PM_DUMP'
+echo >>$cm '#define PCI_HAVE_PM_DUMP'
 echo " dump"
 
 echo_n "Checking for zlib support... "
@@ -133,6 +166,7 @@ else
 	echo >>$c '#define PCI_IDS "pci.ids"'
 fi
 echo >>$c "#define PCI_PATH_IDS_DIR \"$IDSDIR\""
+echo >>$cm "#define PCI_PATH_IDS_DIR \"$IDSDIR\""
 
 echo_n "Checking for DNS support... "
 if [ "$DNS" = yes -o "$DNS" = no ] ; then
@@ -168,4 +202,5 @@ fi
 echo >>$m 'PCILIBPC=$(LIBNAME).pc'
 
 echo >>$c "#define PCILIB_VERSION \"$VERSION\""
-sed '/"/{s/^#define \([^ ]*\) "\(.*\)"$/\1=\2/;p;d;};s/^#define \(.*\)/\1=1/' <$c >>$m
+echo >>$cm "#define PCILIB_VERSION \"$VERSION\""
+sed '/"/{s/^#define \([^ ]*\) "\(.*\)"$/\1=\2/;p;d;};s/^#define \(.*\)/\1=1/' <$cm >>$m
\ No newline at end of file
