--- pciutils-2.1.10/Makefile.strip	2002-03-30 10:42:16.000000000 -0500
+++ pciutils-2.1.10/Makefile	2002-07-18 21:15:59.000000000 -0400
@@ -2,7 +2,7 @@
 # Makefile for Linux PCI Utilities
 # (c) 1998--2002 Martin Mares <mj@ucw.cz>
 
-OPT=-O2 -fomit-frame-pointer
+OPT=-O2
 #OPT=-O2 -g
 CFLAGS=$(OPT) -Wall -W -Wno-parentheses -Wstrict-prototypes
 
@@ -59,7 +59,7 @@
 install: all
 # -c is ignored on Linux, but required on FreeBSD
 	$(DIRINSTALL) -m 755 $(ROOT)/sbin $(PREFIX)/share $(MANDIR)/man8
-	$(INSTALL) -c -m 755 -s lspci setpci $(ROOT)/sbin
+	$(INSTALL) -c -m 755 lspci setpci $(ROOT)/sbin
 	if [ ! -f $(PREFIX)/share/pci.ids -o pci.ids -nt $(PREFIX)/share/pci.ids ] ; then \
 		$(INSTALL) -c -m 644 pci.ids $(PREFIX)/share ; \
 	elif [ -f $(PREFIX)/share/pci.ids ] ; then \
--- pciutils-2.1.9/lspci.c.old	Tue May 28 23:54:03 2002
+++ pciutils-2.1.9/lspci.c	Wed May 29 00:29:39 2002
@@ -1506,7 +1506,7 @@
 	break;
       case 's':
 	if (msg = pci_filter_parse_slot(&filter, optarg))
-	  die("-f: %s", msg);
+	  die("-s: %s", msg);
 	break;
       case 'd':
 	if (msg = pci_filter_parse_id(&filter, optarg))
