diff -up pciutils-2.2.10/update-pciids.sh.wget-timestamping pciutils-2.2.10/update-pciids.sh
--- pciutils-2.2.10/update-pciids.sh.wget-timestamping	2008-09-22 11:31:22.000000000 +0200
+++ pciutils-2.2.10/update-pciids.sh	2008-09-22 11:39:11.102527588 +0200
@@ -33,7 +33,7 @@ if which curl >/dev/null 2>&1 ; then
 	DL="curl -o $DEST.new $SRC"
     ${quiet} && DL="$DL -s -S"
 elif which wget >/dev/null 2>&1 ; then
-	DL="wget -O $DEST.new $SRC"
+	DL="wget -no-timestamping -O $DEST.new $SRC"
 	${quiet} && DL="$DL -q"
 elif which lynx >/dev/null 2>&1 ; then
 	DL="eval lynx -source $SRC >$DEST.new"
